app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/login/LoginFragment.kt

```diff
@@ -1,13 +1,92 @@
 package com.github.alfabravo2013.readyforexams.presentation.login
 
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
 import com.github.alfabravo2013.readyforexams.R
-import com.github.alfabravo2013.readyforexams.databinding.FragmentLandingBinding
+import com.github.alfabravo2013.readyforexams.databinding.FragmentLoginBinding
+import com.github.alfabravo2013.readyforexams.presentation.ToolbarFragment
 
 class LoginFragment : Fragment(R.layout.fragment_login) {
     private val viewModel: LoginViewModel by viewModels()
+    private val binding: FragmentLoginBinding by lazy { FragmentLoginBinding.bind(requireView()) }
 
-    private val binding: FragmentLandingBinding by lazy { FragmentLandingBinding.bind(requireView()) }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
 
+        setUpToolbar()
+
+        binding.loginLoginButton.setOnClickListener {
+            val enteredEmailAddress = binding.loginEmailEditText.text.toString()
+            val enteredPassword = binding.loginPasswordEditText.text.toString()
+            viewModel.setEmailAddress(enteredEmailAddress)
+            viewModel.setPassword(enteredPassword)
+            viewModel.onLoginButtonClick()
+        }
+
+        binding.loginForgotPasswordLink.setOnClickListener {
+            viewModel.onForgotPasswordLinkClick()
+        }
+
+        binding.loginSignupLink.setOnClickListener {
+            viewModel.onSignupLinkClick()
+        }
+
+        viewModel.errorLoginFailed.observe(viewLifecycleOwner) { event ->
```

Comment:
You do not need to observe several events. You can just observe one and then inside it do something like : when { is EVENT -> do this 
is ANOTHER -> do this
}.

Inside your ViewModel you can create a sealed Class called Event { } and have objects or data classes depending on the info that you need to send to the UI . Check this example here: https://github.com/ivandrag/mvvm-rxjava-android/tree/master/app/src/main/java/ro/dragosivanov/ui/users